src/coreclr/gc/gc.cpp

```diff
@@ -43038,10 +43041,10 @@ void GCHeap::SetYieldProcessorScalingFactor (float scalingFactor)
 {
     assert (yp_spin_count_unit != 0);
     int saved_yp_spin_count_unit = yp_spin_count_unit;
-    yp_spin_count_unit = (int)((float)yp_spin_count_unit * scalingFactor / (float)9);
+    yp_spin_count_unit = (int)((float)original_spin_count_unit * scalingFactor / (float)9);
 
-    // It's very suspicious if it becomes 0
-    if (yp_spin_count_unit == 0)
+    // It's very suspicious if it becomes 0 and also, we don't want to spin too much.
+    if ((yp_spin_count_unit == 0) || (yp_spin_count_unit > 32768))
```

Comment:
why max of 32k here?

Reply:
As Per, @Maoni0's comment on https://github.com/dotnet/runtime/pull/68879#issuecomment-1118093244: 

> I would put a cap on the check too 'cause we don't want to spin too much -
> 
> ` if ((yp_spin_count_unit == 0) || (yp_spin_count_unit > 32768))`

This is to hedge our bets against a large spin count.

Reply:
we do 32*number_of_heaps by default. so this is saying if you had 1024 heaps, which would be very many heaps.

Reply:
hmm.. ok